presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousBatchTaskStatusFetcher.java

```diff
@@ -0,0 +1,165 @@
+            Consumer<Throwable> onFail,
+            Codec<Map<TaskId, TaskStatus>> taskListStatusCodec,
+            Duration refreshMaxWait,
+            Executor executor,
+            HttpClient httpClient,
+            Duration maxErrorDuration,
+            ScheduledExecutorService errorScheduledExecutor,
+            RemoteTaskStats stats,
+            boolean isBinaryTransportEnabled) {
+        idWorkerMap = new ConcurrentHashMap<>();
+        workerTaskMap = new ConcurrentHashMap<>();
+
+        this.onFail = requireNonNull(onFail, "onFail is null");
+        this.taskListStatusCodec = requireNonNull(taskListStatusCodec, "taskListStatusCodec is null");
+        this.refreshMaxWait = requireNonNull(refreshMaxWait, "refreshMaxWait is null");
+
+        this.executor = requireNonNull(executor, "executor is null");
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+
+        this.maxErrorDuration = requireNonNull(maxErrorDuration, "maxErrorDuration is null");
+        this.errorScheduledExecutor = requireNonNull(errorScheduledExecutor, "errorScheduledExecutor is null");
+        this.stats = requireNonNull(stats, "stats is null");
+        this.isBinaryTransportEnabled = isBinaryTransportEnabled;
+    }
+
+    public void addTask(
+            TaskId taskId,
+            TaskStatus initialTaskStatus)
+    {
+        Task newTask = new Task();
+        requireNonNull(initialTaskStatus, "initialTaskStatus is null");
+
+        newTask.taskId = requireNonNull(taskId, "taskId is null");
+        newTask.taskStatus = new StateMachine<>("task-" + taskId, executor, initialTaskStatus);
+
+        String worker = initialTaskStatus.getSelf().getHost();
+        idWorkerMap.put(taskId, worker);
+        if (!workerTaskMap.containsKey(worker)) {
+            workerTaskMap.put(
+                    worker,
+                    new WorkerTaskStatusFetcher(
+                            worker,
+                            onFail,
+                            refreshMaxWait,
+                            this.taskListStatusCodec,
+                            executor,
+                            httpClient,
+                            maxErrorDuration,
+                            errorScheduledExecutor,
+                            stats,
+                            isBinaryTransportEnabled
+                    )
+            );
+        }
+        workerTaskMap.get(worker).addTask(newTask);
+    }
+
+    @PostConstruct
+    public synchronized void start()
+    {
+        if (isRunning()) { // We have already called start
+            return;
+        }
+        scheduleNextRequest();
+    }
+
+    public synchronized void stop()
+    {
+        if (future != null) {
+            // do not terminate if the request is already running to avoid closing pooled connections
+            future.cancel(false);
+            future = null;
+        }
+    }
+
+    private synchronized void scheduleNextRequest()
```

Comment:
i am not sure how exactly this going to kickoff workers, given when this method called we won't have any running task. And post us adding task, we are not calling this method. I think we may not need this at all and just putting this logic in addTask method will do the work for us. So whenever we add a new task, it checks if the workerStatusFetcher is running for that worker then no need to do anything else, kick off a new workerTaskStatusFetcher for the new worker.